src/main/webapp/app/overview/course-archive/course-archive.component.html

```diff
@@ -0,0 +1,62 @@
+@if (courses && courses.length) {
```

Comment:
```suggestion
@if (courses) {
```
This should do the same

Reply:
This additional check should be necessary, because if the response is null, we declare courses as an empty array. I just moved this check to another part of the code to display the message of "No courses found or smt similar" properly.